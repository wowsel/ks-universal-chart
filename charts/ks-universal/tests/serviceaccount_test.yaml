suite: test serviceaccount
templates:
  - deployment.yaml
tests:
  - it: should create a service account when autoCreateServiceAccount is enabled
    set:
      deployments:
        test-app:
          containers:
            main:
              image: nginx
              imageTag: latest
          autoCreateServiceAccount: true
          serviceAccountConfig:
            annotations:
              example.com/role: "webserver"
    asserts:
      - matchSnapshot: {} 