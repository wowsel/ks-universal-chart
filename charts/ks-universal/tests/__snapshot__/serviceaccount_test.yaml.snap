should create a service account when autoCreateServiceAccount is enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-test-version
      name: test-app
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: test-app
      template:
        metadata:
          labels:
            app.kubernetes.io/component: test-app
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            helm.sh/chart: ks-universal-test-version
        spec:
          containers:
            - image: nginx:latest
              name: main
          serviceAccountName: test-app
  2: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        example.com/role: webserver
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-test-version
      name: test-app
